src/remote-config/remote-config-api-client.ts

```diff
@@ -160,6 +160,43 @@ export class RemoteConfigApiClient {
       });
   }
 
+  public publishTemplate(template: RemoteConfigTemplate, options?: { force: boolean }): Promise<RemoteConfigTemplate> {
+    let ifMatch: string = template.etag;
```

Comment:
What if `template.etag` is null or undefined and force is not true?

Reply:
If force is not true and `template.etag` is undefined or null the developer will see the following `network-error`:
```
{ code: 'app/network-error',
     message:
      'Error while making request: Invalid value "undefined" for header "If-Match". Error code: ERR_HTTP_INVALID_HEADER_VALUE' },
  codePrefix: 'app' }
```

Reply:
They should see a better error message. We should handle this as an invalid-argument case.

Reply:
For `null` etag we get the following error:
```
{ code: 'remote-config/failed-precondition',
     message:
      '[VERSION_MISMATCH]: The provided ETag [null] is not of the format "etag-<project number>-<version number>".' },
  codePrefix: 'remote-config' }
```

Reply:
I don't mind adding a check for `null` an `undefined` locally if we think that adds any value though

Reply:
Just notice your comment. I will add a check to handle this as an invalid-argument case.

Reply:
Added the check to `sendPutRequest()`